src/ApiConnectors/BaseApiConnector.php

```diff
@@ -169,53 +175,52 @@ protected function getFinderService(): FinderService
 
     /**
      * Convert options array to an ArrayOfString which is accepted by Twinfield.
+     * 
+     * In some cases you are not allowed to change certain options (such as the dimtype, which should always be DEB when using CustomerApiConnector->ListAll()),
+     * in which case the $forcedOptions parameter will be set by the ApiConnector for this option, which will override any user settings in $options
      *
      * @param array $options
-     * @param array|null $forcedOptions
+     * @param array $forcedOptions
      * @return array
      * @throws Exception
      */
-    public function convertOptionsToArrayOfString(array $options, array $forcedOptions = null): array {
+    public function convertOptionsToArrayOfString(array $options, array $forcedOptions = []): array {
         if (isset($options['ArrayOfString'])) {
             return $options;
-        } else {
-            $optionsArrayOfString = array('ArrayOfString' => array());
-
-            if (isset($forcedOptions)) {
-                foreach ($forcedOptions as $key => $value) {
-                    unset($options[$key]);
-                    $optionsArrayOfString['ArrayOfString'][] = array($key, $value);
-                }
-            }
+        }
+        
+        $optionsArrayOfString = ['ArrayOfString' => []];
 
-            foreach ($options as $key => $value) {
-                $optionsArrayOfString['ArrayOfString'][] = array($key, $value);
-            }
+        foreach ($forcedOptions as $key => $value) {
+            unset($options[$key]);
+            $optionsArrayOfString['ArrayOfString'][] = array($key, $value);
+        }
 
-            return $optionsArrayOfString;
+        foreach ($options as $key => $value) {
+            $optionsArrayOfString['ArrayOfString'][] = array($key, $value);
         }
+
+        return $optionsArrayOfString;
     }
 
     /**
      * Map the response of a listAll to an array of the requested class
      *
-     * @param string $className
+     * @param string $objectClass
      * @param $data
      * @param array $objectListAllTags
```

Comment:
This is still missing an explanation of its usage.

Reply:
Done